src/view/com/util/forms/PostDropdownBtn.tsx

```diff
@@ -344,6 +354,78 @@ let PostDropdownBtn = ({
     toggleReplyVisibility,
   ])
 
+  const onPressPin = useCallback(async () => {
```

Comment:
Kinda nit kinda not: we may want to move this into the queries dir as a mutation now just in case we use it in the future elsewhere. But also, we then get `isPending` for free, and since we gotta wait for app view, we may want to add a spinner as the Menu item icon and disable that Menu item, even while the menu is closed (could reopen). Detach quote has this, for an example.